bindings/node/src/mongocrypt.cc

```diff
@@ -528,64 +528,89 @@ Value MongoCrypt::MakeExplicitEncryptionContext(const CallbackInfo& info) {
         throw TypeError::New(Env(), "Parameter `value` must be a Buffer");
     }
 
-    if (info.Length() > 1) {
-        Object options = info[1].ToObject();
+    Object options = info.Length() > 1 ? info[1].ToObject() : Object();
 
-        if (options.Has("keyId")) {
-            Napi::Value keyId = options["keyId"];
+    if (options.Has("keyId")) {
```

Comment:
Does this _work_ when `options` is `Object()`/`nullptr`? I would be surprised if it didn’t throw tbh

Reply:
You're wondering specifically if `Object.has(<key>)` throws on a default object?  I assumed it did not from the documentation:

https://github.com/nodejs/node-addon-api/blob/main/doc/object.md#empty-constructor
https://github.com/nodejs/node-addon-api/blob/main/doc/object.md#has (returns false if not present)

However, I may have mis-interpreted what an "empty object" means, I'll test it and see

Reply:
Yeah, I think it would – “empty” means “null pointer” here, and the Node-API functions should return/throw an error in that case. (`Napi::Object::New()` is the function that creates an empty object in the JS sense, i.e. `{ }`)

Reply:
You're correct - this throws an exception:
```c++
Napi::Boolean Method(const Napi::CallbackInfo& info) {
  const Object obj = Object();
  const bool isPresent = obj.Has("some key");
  Napi::Env env = info.Env();
  return Napi::Boolean::New(env, isPresent);
}
```